tests/src/JIT/opt/FastTailCall/FastTailCallCandidates.cs

```diff
@@ -0,0 +1,1150 @@
+                                           (int) six,
+                                           (int) seven,
+                                           (int) eight,
+                                           first,
+                                           second,
+                                           third,
+                                           fourth,
+                                           fifth,
+                                           sixth); // Cannot fastTailCall
+        }
+        else
+        {
+            return DoNotFastTailCallHelper((int) one,
+                                           (int) two,
+                                           (int) three,
+                                           (int) four,
+                                           (int) five,
+                                           (int) six,
+                                           (int) seven,
+                                           (int) eight,
+                                           first,
+                                           second,
+                                           third,
+                                           fourth,
+                                           fifth,
+                                           sixth); // Cannot fastTailCall
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // HFAs
+    ////////////////////////////////////////////////////////////////////////////
+
+    public struct HFASize16
+    {
+        public double a;
+        public double b;
+
+        public HFASize16(double a, double b)
+        {
+            this.a = a;
+            this.b = b;
+        }
+    }
+
+    public struct HFASize24
+    {
+        public double a;
+        public double b;
+        public double c;
+
+        public HFASize24(double a, double b, double c)
+        {
+            this.a = a;
+            this.b = b;
+            this.c = c;
+        }
+    }
+
+    public struct HFASize32
+    {
+        public double a;
+        public double b;
+        public double c;
+        public double d;
+
+        public HFASize32(double a, double b, double c, double d)
+        {
+            this.a = a;
+            this.b = b;
+            this.c = c;
+            this.d = d;
+        }
+    }
+
+    /// <summary>
+    /// Possible to fastTailCall only on arm64
+    /// </summary>
+    /// <remarks>
+    ///
+    /// This test case is really only interesting on arm64
+    ///
+    /// Arm64:
+    /// caller has 8 register arguments and no stack space
+    /// callee has 8 register arguments and no stack space
+    ///
+    /// x64 *nix:
+    /// caller has 4 register arguments and 64 bytes of stack space
+    /// callee has 8 register arguments
+    ///
+    /// Arm64 can fastTailCall while x64 linux cannot
+    ///
+    /// Return 100 is a pass.
+    /// Return 107 is a failure.
+    ///
+    /// </remarks>
```

Comment:
this comment is a bit confusing since it is the same as the one below (for "caller"), and really you need to consider the caller and callee together.